midenc-compile/src/stages/cross_ctx/lift_exports.rs

```diff
@@ -0,0 +1,170 @@
+    }
+}
+
+fn generate_lifting_function(
+    component_builder: &mut ComponentBuilder,
+    export_id: InterfaceFunctionIdent,
+    export: ComponentExport,
+    diagnostics: &DiagnosticsHandler,
+) -> CompilerResult<ComponentExport> {
+    let export_func_sig = component_builder
+        .signature(&export.function)
+        .ok_or({
+            let message = format!(
+                "Miden CCABI export lifting generation. Cannot find signature for exported \
+                 function {}",
+                export.function
+            );
+            diagnostics.diagnostic(Severity::Error).with_message(message).into_report()
+        })?
+        .clone();
+
+    // get or create the module for the interface
+    let module_id = export_id.interface.full_name;
+    let mut module_builder = component_builder.module(module_id);
+    let cross_ctx_export_sig =
+        flatten_function_type(&export.function_ty, FlatteningDirection::Lift).map_err(|e| {
+            let message = format!(
+                "Miden CCABI export lifting generation. Signature for exported function {} \
+                 requires flattening. Error: {}",
+                export.function, e
+            );
+            diagnostics.diagnostic(Severity::Error).with_message(message).into_report()
+        })?;
+    if needs_transformation(&cross_ctx_export_sig) {
+        let message = format!(
+            "Miden CCABI export lifting generation. Signature for exported function {} requires \
+             lifting. This is not yet supported",
+            export.function
+        );
+        return Err(diagnostics.diagnostic(Severity::Error).with_message(message).into_report());
+    }
+    assert_core_wasm_signature_equivalence(&export_func_sig, &cross_ctx_export_sig);
+
+    let mut builder = module_builder.function(export_id.function, cross_ctx_export_sig.clone())?;
+    let entry = builder.current_block();
+    let params = builder.block_params(entry).to_vec();
+
+    let dfg = builder.data_flow_graph_mut();
+    // import the Wasm core function
+    if dfg.get_import(&export.function).is_none() {
+        dfg.import_function(
+            export.function.module,
+            export.function.function,
+            export_func_sig.clone(),
+        )
+        .map_err(|_e| {
+            let message = format!(
+                "Miden CCABI export lifting generation. lifting function with name {} in module \
+                 {} with signature {export_func_sig:?} is already imported (function call) with a \
+                 different signature",
+                export.function.function, export.function.module
+            );
+            diagnostics.diagnostic(Severity::Error).with_message(message).into_report()
+        })?;
+    }
+    let span = export.function.function.span();
+    let call = builder.ins().exec(export.function, &params, span);
+    // dbg!(&sig);
+    let result = builder.first_result(call);
+    builder.ins().ret(Some(result), span);
```

Comment:
Probably not relevant yet, because we haven't tested with any "interesting" types, but this is where we would need to handle returning results back across the context boundary, so we'll definitely have to flesh out the implementation here when we get into things that interact with memory, or involve representation changes and such.